src/javaRosa.js

```diff
@@ -1523,8 +1407,80 @@ define([
+            this.__callOld();
+
+            var Itext = this.data.javaRosa.Itext,
+                itextMap = this.data.javaRosa.itextMap;
+
+            function getITextID(value) {
+                try {
+                    var parsed = xpath.parse(value);
+                    if (parsed instanceof xpathmodels.XPathFuncExpr &&
+                        parsed.id === "jr:itext")
+                    {
+                        return parsed.args[0].value;
+                    }
+                } catch (err) {
+                    // this seems like a real error since the reference should presumably
+                    // have been valid xpath, but don't deal with it here
+                }
+                return "";
+            }
+
+            function getItextItem(id, property) {
+                var auto = !id || id === getDefaultItextId(mug, property);
+                if (id) {
+                    var item = itextMap[id];
+                    if (item && itextMap.hasOwnProperty(id)) {
+                        if (!auto) {
+                            item.autoId = false;
+                        }
+                        return item;
+                    }
+                }
+                return Itext.createItem(id, auto);
+            }
+
+            function parseItextRef($el, property) {
+                var ref = $el.attr('ref');
+                return getItextItem(ref ? getITextID(ref) : "", property);
+            }
+
+            var labelEl = controlElement.children('label'),
+                hintEl = controlElement.children('hint'),
+                helpEl = controlElement.children('help');
+            if (labelEl.length && mug.spec.label.presence !== 'notallowed') {
+                var labelItext = parseItextRef(labelEl, "label");
+                if (labelItext.isEmpty()) {
+                    //if no default Itext has been set, set it with the default label
+                    var labelVal = xml.humanize(labelEl);
+                    if (labelVal) {
+                        labelItext.setDefaultValue(labelVal);
+                    } else {
+                        // or some sensible deafult
+                        labelItext.setDefaultValue(mug.getDefaultLabelValue());
+                    }
+                }
+                mug.p.labelItext = labelItext;
+            }
+            if (hintEl.length && mug.spec.hintLabel.presence !== 'notallowed') {
+                mug.p.hintItext = parseItextRef(hintEl, "hint");
+            }
+            if (helpEl.length && mug.spec.label.presence !== 'notallowed') {
+                mug.p.helpItext = parseItextRef(helpEl, "help");
+            }
+            if (mug.p.constraintMsgAttr) {
+                var id = getITextID(mug.p.constraintMsgAttr);
+                if (id) {
+                    mug.p.constraintMsgItext = getItextItem(id, "constraintMsg");
+                    mug.p.constraintMsgAttr = null;
+                }
+            }
```

Comment:
All of the above had been in parser.js, intermingled with non-itext label parsing.